overlord/snapstate/snapstate_update_test.go

```diff
@@ -15556,7 +15580,7 @@ func (s *snapmgrTestSuite) TestUpdateWithComponentsRunThroughAdditionalComponent
 
 func (s *snapmgrTestSuite) TestUpdateWithComponentsRunThroughAdditionalComponentsAlreadyInstalled(c *C) {
 	s.testUpdateWithComponentsRunThrough(c, updateWithComponentsOpts{
-		instanceKey:           "key",
+		instanceKey:           "",
```

Comment:
If I understand correctly, is this changed because kernel snaps cannot have instance keys? This does remove some of the cases that we're testing.

Same with `TestInstallInstanceManyComponentsRunThrough` now installing only one component, the intent of that test was to install multiple components.

Reply:
There was an issue with these test as now I had to improve the emulation of kernel updates in the tests. I have done changes now so both kernel and applications snaps using instances are tested.